modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require_dependency 'mobile/application_controller'
+require_relative '../../../models/mobile/v0/adapters/claims_overview'
+require_relative '../../../models/mobile/v0/adapters/claims_overview_errors'
+require_relative '../../../models/mobile/v0/claim_overview'
+
+module Mobile
+  module V0
+    class ClaimsAndAppealsController < ApplicationController
+      include IgnoreNotFound
+      before_action { authorize :evss, :access? }
+
+      def index
+        get_all_claims = lambda {
+          begin
+            claims_list = claims_service.all_claims
+            [claims_list.body['open_claims'].push(*claims_list.body['historical_claims']).flatten, true]
+          rescue => e
+            [Mobile::V0::Adapters::ClaimsOverviewErrors.new.parse(e, 'claims'), false]
+          end
+        }
+
+        get_all_appeals = lambda {
+          begin
+            [appeals_service.get_appeals(@current_user).body['data'], true]
+          rescue => e
+            [Mobile::V0::Adapters::ClaimsOverviewErrors.new.parse(e, 'appeals'), false]
+          end
+        }
+        results = Parallel.map([get_all_claims, get_all_appeals], in_threads: 2, &:call)
+        status_code = parse_claims(results[0], full_list = [], error_list = [])
+        status_code = parse_appeals(results[1], full_list, error_list, status_code)
+        adapted_full_list = serialize_list(full_list.flatten)
+        render json: { data: adapted_full_list, meta: { errors: error_list } }, status: status_code
+      end
+
+      private
+
+      def parse_claims(claims, full_list, error_list)
+        if claims[1]
+          # claims success
+          full_list.push(claims[0].map { |claim| create_or_update_claim(claim) })
+          :ok
+        else
+          # claims error
+          error_list.push(claims[0])
+          :multi_status
+        end
+      end
+
+      def parse_appeals(appeals, full_list, error_list, status_code)
+        if appeals[1]
+          # appeals success
+          full_list.push(appeals[0])
+          status_code
+        else
+          # appeals error
+          error_list.push(appeals[0])
+          status_code == :multi_status ? :bad_gateway : :multi_status
+        end
+      end
+
+      def serialize_list(full_list)
```

Comment:
this looks good and is very clear, but wonder if we can avoid a second iteration through the list by returning the model from the adapter? (I will do same in the appointments adapters) 😄